content/cloud/variables.md

```diff
@@ -0,0 +1,208 @@
+
+This tutorial will walk through creating a simple application that validates a password.
+
+## Prerequisites
+
+First, you need to have Spin installed on your computer. Please use the official Fermyon Cloud Quickstart to both [install](/cloud/quickstart#install-spin) Spin and also [log in](/cloud/quickstart#log-in-to-the-fermyon-cloud) to Fermyon Cloud.
+
+Next, since this example is written in Python, lets make sure we have the required tools installed for Python Spin applications. Spin CLI facilitates the creation of new Spin applications through the use of application [templates](/cloud/cli-reference#templates). The template we are interested in, for this tutorial, is the `http-py` template. Lets ensure its installed along with the `py2wasm` Spin plugin that enables compiling the Python app to Wasm:
+
+<!-- @selectiveCpy -->
+
+```bash
+$ spin templates install --git https://github.com/fermyon/spin-python-sdk --upgrade
+$ spin plugins install py2wasm --yes
+```
+
+The output from the command above will be similar to the following:
+
+<!-- @nocpy -->
+
+```text
+Copying remote template source
+Installing template http-py...
+Installed 1 template(s)
+
++---------------------------------------------+
+| Name      Description                       |
++=============================================+
+| http-py   HTTP request handler using Python |
++---------------------------------------------+
+
+Plugin 'py2wasm' was installed successfully!
+
+Description:
+	A plugin to convert Python applications to Spin compatible modules
+
+Homepage:
+	https://github.com/fermyon/spin-python-sdk
+```
+
+## Creating Our Spin Application
+
+Let's start by [creating a new Spin application](/cloud/cli-reference#new) from a the Python template. Using the docs as a reference, we can perform the following:
+
+<!-- @selectiveCpy -->
+
+```bash
+$ spin new http-py pw_checker
+Project description: A Spin app that validates a password
+HTTP base: /
+HTTP path: /...
+```
+
+## Adding Variables to an Application Component
+
+Next, let's add two variables to our application. Add a top level `[variables]` section to the application manifest (`spin.toml`) and define a `password` and `expected_password` variable, marking them as required. Now, add these variables to a component by referencing them in the `[component.config]` section in the `pw-checker` component. Instead of statically assigning the value of the config variables, we are referencing the dynamic variables with [mustache](https://mustache.github.io/)-inspired string templates. The resultant application manifest should look similar to the following:
```

Comment:
This paragraph would benefit from more motivation - the instructions are clear and easy to follow, but the reader is kind of blindly typing what they're told in the faith that it will all make sense later.  What are these variables?  Why do we need them?  What does the application even do?  (I saw "validates a password" but that's quite open ended.)

I suggest breaking this up a bit more, maybe along the lines of:

> The application receives a password via the URL query string.  It compares that to an expected password, and returns a JSON response indicating whether the submitted password matched or not.
>
> In reality, you'd have multiple user names and password hashes in a database, but for this tutorial we will configure the expected password using a Spin application variable.  We'll call the variable `expected_password`.  There's no reasonable default for it, so we will define it as `required = true`.  To do this, add a `[variables]` section...
>
> The application variable gives us a way to set the expected password, but we need a separate step to surface it to the component.  This separate step is because...  To do this...

Hope this makes sense and sounds reasonable.  It might even benefit from sub-headings to highlight the distinct steps, not sure?

Reply:
I like this a lot. I'll change it to take in the password via the query string